internal/datasources/app_endpoints.go

```diff
@@ -0,0 +1,127 @@
+package datasources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api/app_endpoints"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+)
+
```

Comment:
should verify datasource interface compliance eg https://github.com/couchbasecloud/terraform-provider-couchbase-capella/blob/main/internal/datasources/projects.go#L18